### ISSUE TITLE
Fastify

### ISSUE THREAD

**Issue by C45tr0** [nestjs/swagger#96]
## I'm submitting a...
<pre><code>
[ ] Regression 
[ ] Bug report
[x] Feature request
[ ] Documentation issue or request
[ ] Support request => Please do not submit support request here, instead post your question on Stack Overflow.
</code></pre>

In response to this: https://github.com/nestjs/nest/issues/727. 

To implement this were you thinking of using https://github.com/fastify/fastify-swagger? Or is there another plan? Wondering if I could take a chance at setting something up. Looking for some guidance first though.

**Comment by kamilmysliwiec** [comment]
I'm guessing that we'll use fastify-swagger. Feel free to create a pull request 🙂

**Comment by kamilmysliwiec** [comment]
Fastify integration is available since `v2.2.0` release 🙉❤️

**Comment by Sturgelose** [comment]
@kamilmysliwiec, thanks a lot for the Feature! I gave it a test today and I saw a couple of issues:

1. Locks in the right side (when authorization is enabled for the endpoint) are gone
2. Query parameters (`@Param`) do not appear to be filled in the endpoints
3. @ApiUseTags() seems to not to work as all the endpoints are inside the same group

When rendering them using express, it works as expected. Using exactly the same app/config and just swapping to fastify I can find the previous errors.

Would you like me to create a new issue or maybe this can be tracked in this issue? I guess there must be some differences between the express and fastify implementation of the swagger docs, or I'm not sure...

**Comment by xmlking** [comment]
i also see about issues.  My NPM packages are up to date.

**Comment by kamilmysliwiec** [comment]
We're blocked by this issue https://github.com/fastify/fastify-swagger/issues/97 now.
A possible workaround would be temporarly creating file using generated specification, in this way:

```javascript
const swaggerFile = 'swagger.json';
fs.writeFileSync(swaggerFile, JSON.stringify(document));
httpServer.register(fastifySwagger, {
      swagger: document,
      exposeRoute: true,
      routePrefix: path,
      mode: 'static',
      specification: {
        path: swaggerFile,
      },
});
```

**Comment by kamilmysliwiec** [comment]
My PR to `fastify-swagger` is merged already. `v2.4.2` of `@nestjs/swagger` is available as well.
